File: uni_modules/uts-any-rtc/utssdk/app-android/RtcTypeConverter.kt
package uts.sdk.modules.utsAnyRtc

import com.ss.bytertc.engine.VideoCanvas
import com.ss.bytertc.engine.data.CameraId
import com.ss.bytertc.engine.data.EffectBeautyMode
import com.ss.bytertc.engine.data.MirrorType
import com.ss.bytertc.engine.data.StreamIndex
import com.ss.bytertc.engine.type.ChannelProfile

object RtcTypeConverter {

    fun toChannelProfile(profile: String?): ChannelProfile {
        return when (profile) {
            "communication" -> ChannelProfile.CHANNEL_PROFILE_COMMUNICATION
            "liveBroadcasting" -> ChannelProfile.CHANNEL_PROFILE_LIVE_BROADCASTING
            "game" -> ChannelProfile.CHANNEL_PROFILE_GAME
            "cloudGame" -> ChannelProfile.CHANNEL_PROFILE_CLOUD_GAME
            else -> ChannelProfile.CHANNEL_PROFILE_COMMUNICATION
        }
    }

    fun toStreamIndex(index: String?): StreamIndex {
        return when (index) {
            "main" -> StreamIndex.STREAM_INDEX_MAIN
            "screen" -> StreamIndex.STREAM_INDEX_SCREEN
            else -> StreamIndex.STREAM_INDEX_MAIN
        }
    }

    fun toCameraId(facing: String?): CameraId {
        return when (facing) {
            "front" -> CameraId.CAMERA_ID_FRONT
            "back" -> CameraId.CAMERA_ID_BACK
            else -> CameraId.CAMERA_ID_FRONT
        }
    }

    fun toMirrorType(type: String?): MirrorType {
        return when (type) {
            "none" -> MirrorType.MIRROR_TYPE_NONE
            "render" -> MirrorType.MIRROR_TYPE_RENDER
            "renderAndEncoder" -> MirrorType.MIRROR_TYPE_RENDER_AND_ENCODER
            else -> MirrorType.MIRROR_TYPE_NONE
        }
    }

    fun toRenderMode(mode: String?): Int {
        return when (mode) {
            "hidden" -> VideoCanvas.RENDER_MODE_HIDDEN
            "fit" -> VideoCanvas.RENDER_MODE_FIT
            "fill" -> VideoCanvas.RENDER_MODE_FILL
            else -> VideoCanvas.RENDER_MODE_HIDDEN
        }
    }

    fun toEffectBeautyMode(mode: String): EffectBeautyMode {
        return when (mode) {
            "whiten" -> EffectBeautyMode.WHITE
            "smooth" -> EffectBeautyMode.SMOOTH
            "sharpen" -> EffectBeautyMode.SHARPEN
            "clear" -> EffectBeautyMode.CLEAR
            else -> EffectBeautyMode.SMOOTH
        }
    }
}

File: uni_modules/uts-any-rtc/utssdk/app-ios/RtcTypeConverter.swift
import Foundation
import BytePlusRTC

class RtcTypeConverter {

    static func toChannelProfile(_ profile: String?) -> ByteRTCRoomProfile {
        switch profile {
        case "communication":
            return .communication
        case "liveBroadcasting":
            return .liveBroadcasting
        case "game":
            return .game
        case "cloudGame":
            return .cloudGame
        default:
            return .communication
        }
    }

    static func toStreamIndex(_ index: String?) -> ByteRTCStreamIndex {
        switch index {
        case "main":
            return .indexMain
        case "screen":
            return .indexScreen
        default:
            return .indexMain
        }
    }

    static func toCameraId(_ facing: String?) -> ByteRTCCameraID {
        switch facing {
        case "front":
            return .front
        case "back":
            return .back
        default:
            return .front
        }
    }

    static func toMirrorType(_ type: String?) -> ByteRTCMirrorType {
        switch type {
        case "none":
            return .none
        case "render":
            return .render
        case "renderAndEncoder":
            return .renderAndEncoder
        default:
            return .none
        }
    }

    static func toRenderMode(_ mode: String?) -> ByteRTCRenderMode {
        switch mode {
        case "hidden":
            return .hidden
        case "fit":
            return .fit
        case "fill":
            return .fill
        default:
            return .hidden
        }
    }

    static func toEffectBeautyMode(_ mode: String) -> ByteRTCEffectBeautyMode {
        switch mode {
        case "whiten":
            return .white
        case "smooth":
            return .smooth
        case "sharpen":
            return .sharpen
        default:
            return .smooth
        }
    }
}
